hip/test/base/math.hip.cpp

```diff
@@ -145,4 +145,18 @@ TEST_F(IsFinite, DoubleComplex)
 }
 
 
+// test that zero and one work
+void compilation_test()
+{
+    auto value1 = gko::zero<std::complex<double>>();
+    auto value2 = gko::one<std::complex<double>>();
+}
+
+__global__ void compilation_test2()
+{
+    auto value1 = gko::zero<thrust::complex<double>>();
+    auto value2 = gko::one<thrust::complex<double>>();
+}
```

Comment:
Note: I wouldn't consider these tests essential, as the ParILUT code will trigger the corresponding compilation issues anyways.

Reply:
nit: I think it would be better to make proper tests out of these, like:
```
auto one = gko::zero<thrust::complex<double>>();

ASSERT_EQ(one.real(), 1.0);
ASSERT_EQ(one.imag(), 0.0);
```